### PR TITLE
Fix DiffUNet diffusion

### LINKED ISSUE #359: DiffUnet baselines
Following update from #346 sampling algorithms relying on DiffUnet were not updated properly. This includes DPS and diffpir where normalization is not handled properly

### PR DESCRIPTION
**Pull request by matthieutrs**
Fixing DiffUNet-based algorithms. In a nutshell, normalization in DPS / DiffPIR algorithms were not updated appropriately. Now:
- DPS, DiffPIR work
- DPS, DiffPIR examples work

This fixes #359 

TODO:
- [x] fix DiffPIR
- [x] fix DPS
- [x] add tests for diffusion quality on toy inpainting problems
- [x] fix demo DiffPIR

### Checks to be done before submitting your PR
- [x] `python3 -m pytest tests/` runs successfully.
- [x] `black .` runs successfully.
- [x] `make html` runs successfully (in the `docs/` directory).
- [x] Updated docstrings related to the changes (as applicable).
- [ ] Added an entry to the [CHANGELOG.rst](../CHANGELOG.rst).

### CODE REVIEW

File: examples/sampling/demo_diffpir.py [hunk 1]
```diff
@@ -58,7 +61,7 @@
     imgs,
     titles=["measurement", "ground-truth"],
 )
-
+# print(asdasd)
```
Comment (tachella) [inline examples/sampling/demo_diffpir.py:64]: this seems to have slipped

File: examples/sampling/demo_diffpir.py [hunk 2]
```diff
@@ -265,52 +283,91 @@ def find_nearest(array, value):
...
+        sigma_cur = sigmas[seq[i]]
+
+        # time step associated with the noise level sigmas[i]
+        t_i = find_nearest(reduced_alpha_cumprod, sigma_cur.cpu().numpy())
+        at = 1 / sqrt_recip_alphas_cumprod[t_i] ** 2
+
+        # Denoising step
+        x_aux = x / (2 * at.sqrt()) + 0.5  # renormalize in [0, 1]
+        out = model(x_aux, sigma_cur / 2)
+        denoised = 2 * out - 1
```
Comment (tachella) [inline examples/sampling/demo_diffpir.py:343]:
```suggestion
        denoised = 2 * out - 1 # back to [-1, 1]
```

**Review by tachella — CHANGES_REQUESTED** [review]
the PR looks good, I just found two details to fix, and also please add the fix to the changelog before merging

File: examples/sampling/demo_diffpir.py (hunk 1 above)
Comment (matthieutrs) [reply examples/sampling/demo_diffpir.py]: thanks! done

File: examples/sampling/demo_diffpir.py (hunk 2 above)
Comment (matthieutrs) [reply examples/sampling/demo_diffpir.py]: thanks! done

File: deepinv/tests/test_sampling.py [hunk 3]
```diff
@@ -129,10 +128,57 @@ def test_diffpir(device):
...
     algorithm = DiffPIR(model, likelihood, max_iter=5, verbose=False, device=device)
 
-    out = algorithm(y, physics)
+    with torch.no_grad():
+        out = algorithm(y, physics)
+
     assert out.shape == x.shape
 
 
+def test_diffpir_inpaint(device):
```
Comment (tachella) [inline deepinv/tests/test_sampling.py:137]: merge this test with DPS one using a pytest.parameterize decorator
